core/src/main/scala/org/typelevel/vault/Key.scala

```diff
@@ -30,22 +31,22 @@ import cats.implicits._
   * Since it can only be created as a result of that, it links
   * a Unique identifier to a type known by the compiler.
   */
-final class Key[A] private (private[vault] val unique: Unique) {
+final class Key[A] private (private[vault] val unique: Unique.Token) {
   override def hashCode(): Int = unique.hashCode()
 }
 
 object Key {
   /**
    * Create A Typed Key
    */
-  def newKey[F[_]: Sync, A]: F[Key[A]] = Unique.newUnique[F].map(new Key[A](_))
+  def newKey[F[_]: Sync, A]: F[Key[A]] = Sync[F].unique.map(new Key[A](_))
```

Comment:
All we need here is a `Unique` and a `Functor`.  The `Sync` constraint rules out `Spawn` unless we're also `Async`.  Maybe we should just take two constraints here?

Reply:
good point, https://github.com/typelevel/vault/pull/203/commits/afa2511a5559407cd2a59fc93b33f19ad2a1dd6e